spec/controller/source_data_reader_spec.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require_relative '../../app/controller/source_data_reader'
+
+describe SourceDataReader do
+  before do
+    @reader = SourceDataReader.new('spec/')
```

Comment:
`let(:reader) { SourceDataReader.new('spec/') }`

https://relishapp.com/rspec/rspec-core/docs/helper-methods/let-and-let